patches/minecraft/net/minecraft/world/gen/ChunkProviderServer.java.patch

```diff
@@ -1,14 +1,39 @@
 --- ../src-base/minecraft/net/minecraft/world/gen/ChunkProviderServer.java
 +++ ../src-work/minecraft/net/minecraft/world/gen/ChunkProviderServer.java
-@@ -35,6 +35,7 @@
+@@ -33,8 +33,31 @@
+     private final Set<Long> field_73248_b = Sets.<Long>newHashSet();
+     public final IChunkGenerator field_186029_c;
      public final IChunkLoader field_73247_e;
-     public final Long2ObjectMap<Chunk> field_73244_f = new Long2ObjectOpenHashMap(8192);
+-    public final Long2ObjectMap<Chunk> field_73244_f = new Long2ObjectOpenHashMap(8192);
++    // Forge start - optimize chunk lookups by using last access cache
++    public final Long2ObjectMap<Chunk> field_73244_f = new Long2ObjectOpenHashMap<Chunk>(8192) {
++
++        private static final long serialVersionUID = 1L;
++        protected Chunk lastChunkByPos = null;
++
++        @Override
++        public Chunk get(long key) {
++            if (this.lastChunkByPos != null && key == this.lastChunkByPos.chunkKey) {
```

Comment:
Why not just make this generic and have the class store lastEntry and lastKey? Instead of making this Chunk specific and adding a new field to chunk?